shuttleofx_client/views.py

```diff
@@ -469,60 +470,82 @@ def createUserRepo():
+        os.remove(os.path.join(pluginTemplatePath,'createPlugin.py'))
+        os.remove(jsonFile)
+
+        check_call(['git', 'add', '-A'], cwd=pluginTemplatePath)
+        check_call(['git', 'commit', '--amend', '-m', 'Initial Openfx structure'], cwd=pluginTemplatePath)
+
+        #Create user's repo
+        req = config.github.post('/user/repos', data={
+            "name": request.form['pluginName'],
+            "private": False,
+        }, format='json')
 
         # Push on user's repo
         owner = req.data['owner']['login']
         url = 'https://' + owner + ':' + session['github_token'][0] + '@github.com' +'/'+ owner +'/'+ request.form['pluginName'] + '.git'
-        call(['git', 'remote', 'add', 'origin', url])
-        call(['git', 'remote', '-v'])
-        call(['git', 'push', 'origin', 'master'])
+        check_call(['git', 'remote', 'add', 'origin', url], cwd=pluginTemplatePath)
+        check_call(['git', 'remote', '-v'], cwd=pluginTemplatePath)
+        check_call(['git', 'push', 'origin', 'master'], cwd=pluginTemplatePath)
 
+        # Delete cloned repo
         shutil.rmtree(pluginTemplatePath)
-        # Get back to old dir
-        os.chdir(cwd)
         status = 'success'
-        message = 'Repository created with success, check your Github account !'
-    else :
-        logging.warning('Error while creating a repo : ' + req.raw_data)
+        message = 'Repository created with success, check your Github account!'
+    except CalledProcessError:
         status = 'error'
-        message = req.data['errors'][0]['message']
+        message = 'There was an error creating your repository, please try again.'
+
     return render_template("createPlugin.html", user=user, provider=provider, status=status, message=message)
 
 @config.g_app.route('/create/sources', methods=['POST'])
 def downloadPluginTemplate():
-    call(['git', 'clone', '--recursive', 'https://github.com/tuttleofx/pluginOFX.git'], cwd=os.path.join(os.sep, 'tmp'))
-    pluginTemplatePath = os.path.join(os.sep, 'tmp', 'pluginOFX')
-    # Remove .git folder and files not needed by user using sources
-    shutil.rmtree(os.path.join(pluginTemplatePath,'.git'))
-    os.remove(os.path.join(pluginTemplatePath,'.gitmodules'))
+    pluginTemplatePath = os.path.join(os.sep, 'tmp', 'ofxPluginTemplate')
+    try:
+        check_call(['git', 'clone', 'https://github.com/tuttleofx/ofxPluginTemplate'], cwd=os.path.join(os.sep, 'tmp'))
 
-    jsonForm = repoFormToJSON(request)
+        jsonFile = repoFormToJSON(request)
+        check_call(['python','createPlugin.py', jsonFile, './'], cwd=pluginTemplatePath)
+        check_call(['git','submodule', 'update', '-i'], cwd=pluginTemplatePath)
 
-    shutil.make_archive(pluginTemplatePath, 'zip', pluginTemplatePath)
-    shutil.rmtree(pluginTemplatePath)
+        # Remove .git folder and files not needed by user using sources
+        shutil.rmtree(os.path.join(pluginTemplatePath,'.git'))
```

Comment:
No reason to do that, we can do the same than for the github repo.
The only difference is that the "git submodule update -i" is needed to create the archive.